### PR TITLE
Clean domain icons

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add `ha-domain-icon` icon and remove (almost) all icon rules from the frontend repo.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/media-browser/ha-bar-media-player.ts [hunk 1]
```diff
@@ -382,6 +377,23 @@ export class BarMediaPlayer extends SubscribeMixin(LitElement) {
...
+    }
+    if (stateObj) {
+      return html`
+        <ha-state-icon .hass=${this.hass} .stateObj=${stateObj}></ha-state-icon>
+      `;
+    }
+    return html`
+      <ha-domain-icon
+        .hass=${this.hass}
+        domain=${computeDomain(this.entityId)}
```
Comment (bramkragten) [inline src/panels/media-browser/ha-bar-media-player.ts:392]:
```suggestion
        .domain=${computeDomain(this.entityId)}
```

File: src/panels/config/automation/add-automation-element-dialog.ts [hunk 2]
```diff
@@ -318,17 +317,15 @@ class DialogAddAutomationElement extends LitElement implements HassDialog {
             (!domainUsed && manifest?.integration_type === "entity") ||
             !["helper", "entity"].includes(manifest?.integration_type || "")))
       ) {
-        const icon = domainIconWithoutDefault(domain);
         result.push({
           group: true,
-          iconPath: icon,
-          image: !icon
```
Comment (bramkragten) [inline src/panels/config/automation/add-automation-element-dialog.ts:325]: Why was brand image removed?

Comment (piitaya) [reply src/panels/config/automation/add-automation-element-dialog.ts:325]: The logic has been moved into ha-domain-icon.

File: src/components/ha-domain-icon.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    if (this.brandFallback) {
+      const image = brandsUrl({
+        domain: this.domain!,
+        type: "icon",
+        darkOptimized: this.hass.themes?.darkMode,
+      });
+      return html`
+        <img
+          alt=""
+          slot="start"
```
Comment (bramkragten) [inline src/components/ha-domain-icon.ts:62]:
```suggestion
```

**Review by bramkragten — APPROVED** [review]

**Comment by karwosts** [comment]
I can see my precipitation icons have changed from a raincloud to just a generic eyeball. Is that intended? I have latest dev from core and frontend.

**Pull request merged**